pageserver/src/remote_storage/storage_sync/index.rs

```diff
@@ -109,23 +113,49 @@ impl RemoteTimelineIndex {
     pub fn branch_files(&self, tenant_id: ZTenantId) -> Option<&HashSet<RelativePath>> {
         self.branch_files.get(&tenant_id)
     }
+
+    pub fn set_awaits_download(
+        &mut self,
+        id: &ZTenantTimelineId,
+        awaits_download: bool,
+    ) -> anyhow::Result<()> {
+        self.timeline_entry_mut(id)
+            .ok_or_else(|| anyhow::anyhow!("unknown timeline sync {}", id))?
+            .set_awaits_download(awaits_download);
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct DescriptionTimelineIndexEntry {
+    pub description: BTreeMap<ArchiveId, ArchiveDescription>,
+    pub awaits_download: bool,
```

Comment:
I wonder, if we need more than this status in general?
For instance, cases when the timeline was evicted on the error threshold from the sync queue, we can remember that and disallow further synchronization?

NIT: Also feels like we could move this one level up in a parent enum, to ease its access?

Reply:
> Also feels like we could move this one level up in a parent enum, to ease its access?

Done, thanks!

> was evicted on the error threshold

I think if we decide to go with this eviction marker it should be placed in timeline metadata, so this eviction marker can survive restart.

> of the same checkpoint

Indeed this behavior is not desirable, but it should be harmless. It would also be different from awaits download because awaits download is scoped for a timeline, not for a particular archive, but for checkpoints it should be ok to have more than one in a queue, so IMO we shouldn't restrict this behavior. I do not see particular case when this could happen either, so I would rather keep it outside from the scope for this PR if you agree with that

Reply:
Sure.

Let's have the python test reenabled as a proof that we're able to use S3 normally (to enable and test real S3 without major issues) and the rest can come any time later.

Reply:
tests are reenabled and green, eviction or other solution to critical error handling is added to #977 as a follow up